File: src/pagePackage/SelectableCoursesPage.java
package pagePackage;

import java.util.ArrayList;

import systemMessagePackage.FunctionType;
import systemMessagePackage.SystemMessage;

public class SelectableCoursesPage extends Page{
	
	private int numberOfSelectableCourses;
	
	public SelectableCoursesPage(String content) {
		super(content);
		setType(PageType.SELECTABLE_COURSES_PAGE);
		setName("Selectable Courses Page");
	}
	
	@Override
	public SystemMessage runPage() {
		showContent();
		String input;
		
		while(numberOfSelectableCourses > 0) {
			input = takeInput();
			
			try {
				int selection = Integer.parseInt(input);
				
				if (selection > numberOfSelectableCourses && selection < 0) {
					System.out.println("Enter a valid number");
					return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.SELECTABLE_COURSES_PAGE, null);
				}
				else {
					return new SystemMessage(FunctionType.SELECT_COURSE, PageType.SELECTABLE_COURSES_PAGE, selection);
				}
			} 
			catch (NumberFormatException e) {
				
				if (input.equals("q")) {
					return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.MAIN_MENU_PAGE_STUDENT, null);
				}
				else {
					System.out.println("Wrong Input");
					return new SystemMessage(FunctionType.NONE, null, null);
				}
			}	
		}
		
		input = takeInput();
		if (input.equals("q")) {
			return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.MAIN_MENU_PAGE_STUDENT, null);
		}
		else {
			System.out.println("Wrong Input");
			return new SystemMessage(FunctionType.NONE, null, null);
		}		
	}

	
	/////////// GETTER - SETTER METHODS ////////////
	
	public int getNumberOfSelectableCourses() {
		return numberOfSelectableCourses;
	}

	public void setNumberOfSelectableCourses(int numberOfSelectableCourses) {
		this.numberOfSelectableCourses = numberOfSelectableCourses;
	}
}

File: iter3/python_development/Page/SelectableCoursesPage.py
from iter3.python_development.Page.Page import Page
from iter3.python_development.Page.PageType import PageType
from iter3.python_development.System.FunctionType import FunctionType
from iter3.python_development.System.SystemMessage import SystemMessage


class SelectableCoursesPage(Page):

    numberOfSelectableCourses = None

    def __init__(self, content):
        super().__init__(content)
        self.setType("SELECTABLE_COURSES_PAGE")
        self.setName("Selectable Courses Page")


    def runPage(self):
        self.showContent()
        inputFromPage = None
        while(True):
            inputFromPage = self.takeInput()
            try:
                selection = int(inputFromPage)
                if (selection > self.numberOfSelectableCourses or selection < 0):
                    print("Enter a valid number")
                else: 
                    return SystemMessage("SELECT_COURSE", "SELECTABLE_COURSES_PAGE", selection)
            except:
                if (inputFromPage.lower() == "q"):
                    return SystemMessage("CHANGE_PAGE", "MAIN_MENU_PAGE", None)
                else:
                    print("Wrong Input!")


    def getNumberOfSelectableCourses(self):
        return self.numberOfSelectableCourses
    
    def setNumberOfSelectableCourses(self, numberOfSelectableCourses):
        self.numberOfSelectableCourses = numberOfSelectableCourses